src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -851,7 +851,6 @@ private AbstractFrame getMethodWithoutThis(DetailAST ast) {
         if (!validateOnlyOverlapping) {
             final AbstractFrame frame = findFrame(ast, true);
             if (frame != null
-                    && ((ClassFrame) frame).hasInstanceMethod(ast)
                     && !((ClassFrame) frame).hasStaticMethod(ast)) {
```

Comment:
I am having issues determining if this mutation is valid or not and how to prove it.

I will recheck with regression with `validateOnlyOverlapping` = false.

Reply:
This has been reverted.